src/haz3lweb/Main.re

```diff
@@ -102,17 +113,37 @@ module App = {
       (
         model: Incr.t(Haz3lweb.Model.t),
         ~old_model as _: Incr.t(Haz3lweb.Model.t),
-        ~inject,
+        ~inject: UpdateAction.t => Ui_effect.t(unit),
       ) => {
     open Incr.Let_syntax;
     let%map model = model;
     /* Note: mapping over the old_model here may
        trigger an additional redraw */
+    let first = ref(false);
+    let go = ref(false);
```

Comment:
We listen on frame refresh for this mutable value to be set and trigger the action on it being set. Definitely a gross hack but I'm not sure the correct way to deal with external js triggered events.